pkg/disk/attach_helper.go

```diff
@@ -0,0 +1,36 @@
+package disk
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+)
+
+func getDevices() []string {
+	devices := []string{}
+	files, _ := ioutil.ReadDir("/dev")
+	for _, file := range files {
+		if !file.IsDir() && strings.HasPrefix(file.Name(), "vd") {
+			devices = append(devices, fmt.Sprintf("/dev/%s", file.Name()))
+		}
+	}
+
+	return devices
+}
+
+func calcNewDevices(old, new []string) []string {
```

Comment:
@jia-jerry   Just a question: why not simply use the field `Device` of http://docs.aliyun.com/#/pub/ecs/open-api/datatype&diskitemtype

Reply:
Aliyun doesn't always return correct device. You may refer to @denverdino. He said it is an kernal issue and can't fix. 

As a result, the implementation of CSI is awkward. We can't simple use ControllerPublishVolume simple to attach instance. Never try this way.

Reply:
I have concern if there is issue when concurrent attach occurs for the same ecs instance. any comment?